### ISSUE TITLE
bug: Assistive technologies, such as screen readers, can't interpret ARIA attributes with invalid values 

### ISSUE THREAD

**Issue by Narenvaran** [ionic-team/ionic-framework#23182]
# Bug Report

**Ionic version:**
 
[x] 5.4.16

**Current behavior:**

  in ion-input tag aria value do not have valid values eg: **aria-labelledby="ion-input-0-lbl"**
**Expected behavior:**

  It should be proper user-specified value  
**Steps to reproduce:**

Create an ion-input and try assigning an aria-label then run the accessibility testing tool axe (can use google chrome extension).
**Related code:**
 <ion-item class="ion-margin-top item-label input-bg">
            <ion-icon name="calculator-outline"></ion-icon>
            <ion-input name="calculator-outline" id="nameofthe "placeholder='{{ "loginPage.organisationID" | translate }}' [(ngModel)]="organisationID"></ion-input>
          </ion-item>

```
insert short code snippets here
 <ion-item class="ion-margin-top item-label input-bg">
            <ion-icon name="calculator-outline"></ion-icon>
            <ion-input name="calculator-outline" id="nameofthe "placeholder='{{ "loginPage.organisationID" | translate }}' [(ngModel)]="organisationID"></ion-input>
          </ion-item>
```

**Other information:**

**Ionic info:**

```

Ionic:

   Ionic CLI                     : 5.4.16 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.6.3
   @angular-devkit/build-angular : 0.1101.4
   @angular-devkit/schematics    : 11.1.4
   @angular/cli                  : 11.1.4
   @ionic/angular-toolkit        : 3.1.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 8 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 1.3.0

System:

   Android SDK Tools : 26.1.1 
  ios-deploy        : 1.9.4
   NodeJS            : v14.15.3 (/usr/local/bin/node)
   npm               : 6.14.9
   OS                : macOS Catalina
   Xcode             : Xcode 11.6 Build version 11E708

```

**Comment by liamdebeasi** [comment]
Thanks, I can reproduce this. The issue here is that we expect an `ion-label` to be used with `ion-input`, and the `aria-labelledby` ID refers to that `ion-label`. In the event that users don't use `ion-label`, they should put their own `aria-label` attribute.

In v5.6.4 we shipped a fix that let the `aria-label` be inherited to the inner `input` element, but we still add `aria-labelledby` (https://github.com/ionic-team/ionic-framework/commit/61f094d30665c9afec428028883a5d9a085892d8)

--------

Ionic Team: 

To fix this, we should update the code to only add `aria-labelledby` if a valid `label` element was found: https://github.com/ionic-team/ionic-framework/blob/master/core/src/components/input/input.tsx#L411

Something like the following:

```jsx
aria-labelledby={label ? labelId : null}
```

We should apply this fix to `ion-input`, `ion-textarea`, and `ion-datetime` and write tests. While we're at it, we should make sure `aria-label` can be inherited in `ion-textarea` as well.

**Comment by willmartindev** [comment]
Thanks for the issue @Narenvaran! This has been resolved via #23211, and a fix will be available in an upcoming release of Ionic Framework.
